torch/_inductor/utils.py

```diff
@@ -2776,6 +2779,39 @@ def triton_version_uses_attrs_dict() -> bool:
     return get_triton_attrs_descriptor_version() == TritonAttrsDescriptorVersion.V4_DICT
 
 
+def register_custom_op_support_cudagraph(
```

Comment:
Proposal: (sorry I changed my mind):
- the default behavior is that all operators are assumed cudagraphs safe. This is for BC reasons.
- add a tag torch.Tag.cudagraphs_unsafe. The presence of the tag means that the operator is cudagraphs unsafe. The negative name is a little annoying but we've got constraints here.
- add a generic mechanism to set and unset operator tags. opoverload.unsafe_set_tags(...), opoverload.unsafe_unset_tags(...)
- use the generic mechanism to implement some new APIs. The two APIs that I want are:

1) torch.library.register_supports_cudagraphs(op: Union[OpOverload, CustomOpDef])
2) customopdef.register_supports_cudagraphs(True/False)

Reply:
@eellison do you think any operators will support cudagraphs conditionally? The tag-based approach might not make as much sense if so. For example, Boyuan pointed me at https://github.com/pytorch/pytorch/blob/039ebdc19287bba56fdb6d6bb9e693b3c88de927/torch/_inductor/utils.py#L834-L850 where built-in operators conditionally support CUDAGraphs depending on torch.are_deterministic_algorithms_enabled()

Reply:
Can we wait until we have a use case for oss to add a conditional api ?